src/inmanta/env.py

```diff
@@ -59,7 +59,22 @@ class PackageNotFound(Exception):
 
 
 class ConflictingRequirements(Exception):
-    pass
+    def __init__(
+        self, msg: Optional[str] = None, conflicts: Optional[List[Tuple[Requirement, Optional[version.Version]]]] = None
+    ):
+        self.conflicts = conflicts
+        self.msg = msg
+
+    def get_msg(self) -> str:
+        if self.msg:
+            return self.msg
+        msg: str = "Module dependency resolution conflict: a module dependency constraint \
+was violated by another module. This most likely indicates an incompatibility between \
+two or more of the installed modules. You can get more details on the issue with 'pip check'.\n"
+        if self.conflicts is not None:
+            for constraint, v in self.conflicts:
+                msg += "Incompatibility between constraint %s and installed version %s\n" % (constraint, v)
+        return msg
```

Comment:
I wouldn't reference the `pip check` command anymore, as it might not detect the version conflict depending on whether the conflict involves a V1 module or not.

```suggestion
        msg: str = "Module dependency resolution conflict: a module dependency constraint \
was violated by another module. This most likely indicates an incompatibility between \
two or more of the installed modules."
        if self.conflicts is not None:
            for constraint, v in self.conflicts:
                msg += "\n\t* Incompatibility between constraint %s and installed version %s" % (constraint, v)
        return msg
```